File: lib/geometry2/source/revolved_surface_test.py
## \file revolved_surface_test.py
## \ingroup libgeom2
## \brief Exercise the RevolvedSurface class.
## \author PJ
## \version 28-Jan-2006

from math import *
from libgeom2 import *

print "\n\n-----------------------------------------------"
print "Begin revolved_surface_test..."

print "\nSimple RevolvedSurface describing a sphere-cone:"
Rnose = 1.0
Angle = 45.0 * pi / 180.0
Length = 1.0
c = Vector(0.0, 0.0, 0.0)    # centre of radius
a = Vector(-Rnose, 0.0, 0.0)  # tip of nose
b = Vector(-Rnose*cos(Angle), Rnose*sin(Angle), 0.0) # join between sphere and cone
d = b + Length * Vector(cos(Angle), sin(Angle), 0.0) # skirt of cone
path = Polyline([Arc(a,b,c), Line(b,d)])
surf1 = RevolvedSurface(path, "REVOLVED_SURFACE")
print surf1
print "surf1.eval(0.25,0.75)=", surf1.eval(0.25,0.75)

print "\nMapped surface:"
p0 = Vector3(-0.5, -0.5, -0.5)
p1 = Vector3(-0.5, -0.5, 0.5)
p2 = Vector3(-0.5, 0.5, 0.5)
p3 = Vector3(-0.5, 0.5, -0.5)
qsurf = CoonsPatch(p0, p1, p2, p3, "QUERY_SURFACE")
msurf = MappedSurface(qsurf, surf1)
print "msurf=", msurf
print "msurf.eval(0.25,0.75)=", msurf.eval(0.25,0.75)

print "Render to VRML"
outfile = open("revolved_surface_test_py.wrl", "w")
outfile.write("#VRML V2.0 utf8\n")
outfile.write(surf1.vrml_str() + "\n")
outfile.write(msurf.vrml_str() + "\n")
outfile.close()

print "Done."

File: lib/geometry2/source/revolved_surface_test.cxx
/** \file revolved_surface_test.cxx
 *  \ingroup libgeom2
 *  \brief Exercise the RevolvedSurface and MappedSurface classes.
 *  \author PJ
 *  \version 28-Jan-2006
 *
 */
#include <iostream>
#include <fstream>
#include <math.h>
#include <vector>
#include "geom.hh"
#include "gpath.hh"
#include "surface.hh"
using namespace std;


int main()
{
    cout << "\n\n-----------------------------------------------" << endl;
    cout << "Begin revolved_surface_test..." << endl;

    cout << "\nSimple RevolvedSurface:" << endl;
    Line path = Line(Vector3(0.0,0.0,0.0), Vector3(1.0,1.0, 0.0));
    RevolvedSurface surf1 = RevolvedSurface(&path, "REVOLVED_SURFACE");
    cout << "surf1=" << surf1 << endl;
    cout << "surf1.eval(0.25,0.75)=" << surf1.eval(0.25,0.75) << endl;

    cout << "\nMapped surface:" << endl;
    Vector3 p0(-0.5, -0.5, -0.5);
    Vector3 p1(-0.5, -0.5, 0.5);
    Vector3 p2(-0.5, 0.5, 0.5);
    Vector3 p3(-0.5, 0.5, -0.5);
    CoonsPatch qsurf = CoonsPatch(p0, p1, p2, p3, "QUERY_SURFACE");
    MappedSurface msurf = MappedSurface(&qsurf, &surf1);
    cout << "msurf=" << msurf << endl;
    cout << "msurf.eval(0.25,0.75)=" << msurf.eval(0.25,0.75) << endl;

    return 0;
}
